frontend/src/app/store/reducers/cases.ts

```diff
@@ -207,6 +311,61 @@ const updateAssessment =
         dispatch(setAssessment({ assessment: updatedAssessmentData }));
       });
     }
+    const updatePreventionEducation =
+    (
+      complaintIdentifier: string,
+      caseIdentifier: string,
+      preventionEducation: PreventionEducation
+    ): AppThunk =>
+      async (dispatch, getState) => {
+  
+        const {
+          codeTables: { "assessment-type": preventionEducationType },
+          officers: { officers },
+          app: { profile },
+        } = getState();
+  
+        let updatePreventionEducationInput = {
+          updatePreventionEducationInput: {
+            leadIdentifier: complaintIdentifier,
+            caseIdentifier: caseIdentifier,
+            updateUserId: profile.idir_username,
+            agencyCode: "COS",
+            caseCode: "HWCR",
+            preventionEducationDetails: {
+              actions: preventionEducation.prevention_education_type.map((item) => {
+                return {
+                  actor: preventionEducation.officer?.value,
+                  date: preventionEducation.date,
+                  actionCode: item.value,
+                  activeIndicator: true
+                }
+              })
+            }
+          }
+        } as UpdatePreventionEducationInput;
+        let { updatePreventionEducationInput: { preventionEducationDetails: { actions } } } = updatePreventionEducationInput;
+  
+        for (let item of preventionEducationType.filter((record) => record.isActive)) {
```

Comment:
I mentioned this to Dmitri as well, but this for loop should really be simplified, especially becasue you're doing a map and includes in your if statement. 

```
const types = preventionEducationType.filter((record) => record.isActive)

for(let item of types) {
  const hasActionCode = actions.some((action) => action.actionCode === item.assessmentType); 

if (!hasActionCode) {
    actions.push({
      date: preventionEducation.date,
      actor: preventionEducation.officer?.value,
      activeIndicator: false,
      actionCode: item.assessmentType,
    } as AssessmentActionDto);
  }
}
```

this is a lot easier to read and understand